api/dataprocessor.go

```diff
@@ -166,22 +147,22 @@ func aggEvery(numPoints, maxPoints uint32) uint32 {
 }
 
 // error is the error of the first failing target request
-func getTargets(store mdata.Store, reqs []Req) ([]Series, error) {
-	seriesChan := make(chan Series, len(reqs))
+func (s *Server) getTargets(reqs []models.Req) ([]models.Series, error) {
+	seriesChan := make(chan models.Series, len(reqs))
 	errorsChan := make(chan error, len(reqs))
 	// TODO: abort pending requests on error, maybe use context, maybe timeouts too
 	wg := sync.WaitGroup{}
 	wg.Add(len(reqs))
 	for _, req := range reqs {
-		go func(wg *sync.WaitGroup, req Req) {
+		go func(wg *sync.WaitGroup, req models.Req) {
 			pre := time.Now()
-			points, interval, err := getTarget(store, req)
+			points, interval, err := s.getTarget(req)
 			if err != nil {
 				errorsChan <- err
```

Comment:
actually no. let me first give some historical context:

various functions call panic.  this includes divide(), getSeries(), AggMetric Get() and GetAggregated() and consolidation. some of these are critical assertions pointing out severe bugs in MT, others represent "normal" operational errors (like cassandra query errors).

because it's such a nested and wide call hierarchy (getTargets -> getTarget -> getSeries, divide,  etc) adding `if err := foo(); err != nil {return foo, err {` would be a lot of code in many places, so i opted to just panic where needed and call recover in getTarget(), turning it back into normal errors, so we can handle them cleanly to the user.

However by removing doRecover() from getTarget, any such panic (including the ones we write, but also runtime panics!) will crash the runtime.  It looks like the only place where we now recover is the macaron recovery handler, which runs in different goroutines then the goroutines we start in getTargets(), so any panics in the latter will crash metrictank.

What is really needed here is a more proper refactoring that uses contexts so that a panic anywhere in the stack of a getTarget() routine cancels all other work relating to the same request.  But I think that deserves a PR by itself, so 
my suggestion would be to put the doRecover() stuff back for now.